lib/galaxy/datatypes/data.py

```diff
@@ -743,6 +769,9 @@ def dataprovider(self, dataset, data_format, **settings):
             return self.dataproviders[data_format](self, dataset, **settings)
         raise dataproviders.exceptions.NoProviderAvailable(self, data_format)
 
+    def validate(self, dataset, **kwd):
+        return DatatypeValidation.unvalidated()
```

Comment:
Would it make sense to perhaps do a minimum check and make sure that that datatypes' sniffer doesn't return False / raise an exception ?

Reply:
I've got problems with this... although I know we do this in some ways already. I wish the sniffer had a way of saying - I don't know. Because the way it is now I think it is totally fine for a sniffer to report false positives or false negatives - it is an estimate. The validation wouldn't be perfect of course, but I'd like it to be considered a bug if the validation reports false positives or negatives but not the sniffer. I want to start with the assumption that validation is a stronger claim, and I'd like to work toward the sniffer being a weaker claim.

@nsoranzo recommended the same thing though... maybe I'm wrong and we should just make that change and then go through and do the work of tweaking it toward my goals from that basis.

Reply:
I vote for full checks with the validator. Otherwise, won't help as much. 

No one runs _production quality_ line-command bioinfo analysis without fully checking the inputs first -- is the top way to prevent bad data from presenting as "tool errors" when is really an input content problem. Super annoying and wastes resources to discover that after investing a bunch of time/effort/compute to complete upstream work, that sends the result to a downstream tool that is more "picky" about inputs, and fails out later. The solution is to fix the input and rerun the entire thing again. IF the user is even able to figure out that is the root problem (many don't -- and the fixes are usually pretty simple).

SO, investing in complete, early detection of input problems is worth it imo, and the primary goal that at least I had in mind when discussing why to add this in to begin with.  

But maybe I am missing something? Am looking at this from a "how to make analysis more likely to be successfull for the end user" perspective.